### PR TITLE
Add search plugin

### LINKED ISSUE #79: Add search plugin
We should use the search plugin https://handsontable.com/docs/6.2.2/demo-searching.html and make it possible to use a text input widget for searching

### PR DESCRIPTION
**Pull request by martinRenou**
Fix #79

### CODE REVIEW

**Comment by maartenbreddels** [comment]
Really working well, the only thing was when the sheet got created (like you pass it on contruction, or reload the page) it would not highlight it, which I fixed.

**Comment by martinRenou** [comment]
Another bug: when you edit one of the cells, we should maybe trigger another search. But I guess it should be lazy... So that if you are not actually searching for something it does not trigger it. Something like this in the `onchange` event:
```javascript
if (this.model.search_token != "") {
    this._search();
}
```

**Comment by maartenbreddels** [comment]
Good catch, can we write tests for all this? Can we inspect somehow that something is highlighted?

**Comment by martinRenou** [comment]
Maybe through the CSS classes?

**Comment by martinRenou** [comment]
Adding tests right now

File: js/src/test/test_sheet.ts [hunk 1]
```diff
@@ -230,4 +234,33 @@ describe('sheet', function() {
...
+    it('should search after change', async function() {
+        var cell = await make_cell.apply(this, [{value: [['Yop']]}]);
+        var view = await make_view.call(this)
+        await view._table_constructed;
+
+        this.sheet.set('search_token', 'Hell');
+        expect(view.el.querySelector('td[class*="htSearchResult"]')).to.equal(null);
+        console.log(cell.get('value'));
+        cell.set('value', [['Hello']]);
+        await wait(500);
```
Comment (martinRenou) [inline js/src/test/test_sheet.ts]: I don't like this. But could not find a way around it. @maartenbreddels is there a way to wait for all the events/callbacks to be done after the `set` call?

**Pull request merged**